include/permanent/ryser.h

```diff
@@ -95,120 +126,72 @@ namespace permanent {
 template <typename T, typename I = void>
 result_t<T, I> ryser_square(const size_t m, const size_t n, const T *ptr)
 {
-  (void)n;
-
-  size_t i, j;
-  size_t k;
-  result_t<T, I> rowsum;
-  result_t<T, I> out = 0;
-
-  /* Iterate over c = pow(2, m) submatrices (equal to (1 << m)) submatrices.
-   */
-
-  size_t c = 1UL << m;
-
-  /* Loop over columns of submatrix; compute product of row sums. */
-
-  for (k = 0; k < c; ++k) {
-    result_t<T, I> rowsumprod = 1.0;
-    for (i = 0; i < m; ++i) {
-      /* Loop over rows of submatrix; compute row sum. */
-
-      rowsum = 0.0;
-      for (j = 0; j < m; ++j) {
-        /* Add element to row sum if the row index is in the
-         * characteristic * vector of the submatrix, which is the binary
-         * vector given by k.  */
-
-        if (k & (1UL << j)) {
-          rowsum += ptr[m * i + j];
+    (void)n;
+
+    size_t i, j, k;
```

Comment:
Initialize variables at the deepest scope possible. Like in the for loops, `for (size_t i = 0; .....)`. Check this generally. I cleaned it up most other places.